golang/go-cloud-run-hello-world/main.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+
+	"cloud.google.com/go/compute/metadata"
```

Comment:
How this will work when the template will be running locally?

Reply:
Current behavior is wait for timeout then it falls back to default `???`.
We could preempt the metadata server request by checking for a `GOOGLE_CLOUD_PROJECT` env var. Getting the Project ID on Cloud Run has a metadata server dependency.

Reply:
I like that!

Check the GOOGLE_CLOUD_PROJECT env var, if it's set, wait for timeout and fall back if needed. If it's not set, it means you run locally.

Reply:
Done.